app/components/Currency/index.tsx

```
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { FormattedNumber } from 'react-intl';
+
+interface CurrencyProps {
+    type?: string;
+    value: (number | string);
+}
+
+const Currency: React.FC<CurrencyProps> = (props: CurrencyProps) => {
+    const {
+        type = 'USD',
+        value,
+    } = props;
+
+    const amount = (value && !isNaN(+value)) ? value : '0';
```

Should we throw an error if a non-numeric value is used instead of justing setting it to 0?

---

Interesting question.  Let's leave it for now and we can address it if it becomes a problem later.

---

actually the value could be a number or a string and it would be handled by the FormattedNumber component